From ec13470c08160161c30d4fe1cdae20122c1438ab Mon Sep 17 00:00:00 2001
Message-Id: <patch-v8-01.12-ec13470c081-20220118T155150Z-avarab@gmail.com>
In-Reply-To: <cover-v8-00.12-00000000000-20220118T155150Z-avarab@gmail.com>
References: <cover-v7-00.10-00000000000-20211228T004707Z-avarab@gmail.com>
	<cover-v8-00.12-00000000000-20220118T155150Z-avarab@gmail.com>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Tue, 2 Feb 2021 10:54:38 +0100
Subject: [PATCH v8 01/12] grep.h: make patmatch() a public function
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

I'd like to use the PCRE & ERE etc. code in grep.c for more things in
git, starting with diffcore-pickaxe.c.

The current API just exposes grep_{source,buffer}() for that
purpose. I could use those, but they're very fat entry points into the
entire set of bells and whistles that grep.c supports for "git
grep". I just want the equivalent of a light regexec() wrapper for my
compiled patterns.

So let's expose patmatch() for that purpose. It's not perfect, in
particular it's a bit ugly that we need to pop a pattern off the
opt->pattern_list if all we've got is the "grep_opt" wrapper struct,
but it'll do for now.

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 grep.c | 6 +++---
 grep.h | 3 +++
 2 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/grep.c b/grep.c
index 3497df48ca6..be4bbc4b7c4 100644
--- a/grep.c
+++ b/grep.c
@@ -843,9 +843,9 @@ static void show_name(struct grep_opt *opt, const char *name)
 	opt->output(opt, opt->null_following_name ? "\0" : "\n", 1);
 }
 
-static int patmatch(struct grep_pat *p,
-		    const char *line, const char *eol,
-		    regmatch_t *match, int eflags)
+int patmatch(struct grep_pat *p,
+	     const char *line, const char *eol,
+	     regmatch_t *match, int eflags)
 {
 	int hit;
 
diff --git a/grep.h b/grep.h
index 460cb75a357..8890ce2d25f 100644
--- a/grep.h
+++ b/grep.h
@@ -244,6 +244,9 @@ void grep_source_load_driver(struct grep_source *gs,
 
 
 int grep_source(struct grep_opt *opt, struct grep_source *gs);
+int patmatch(struct grep_pat *p,
+	     const char *line, const char *eol,
+	     regmatch_t *match, int eflags);
 
 struct grep_opt *grep_opt_dup(const struct grep_opt *opt);
 
-- 
2.35.0.rc1.864.g57621b115b6

